domain/src/test/java/org/oppia/domain/topic/StoryProgressControllerTest.kt

```diff
@@ -527,22 +525,30 @@ class StoryProgressControllerTest {
 
   private fun verifyTopicProgressFailed() {
     verify(mockTopicProgressObserver, atLeastOnce()).onChanged(topicProgressResultCaptor.capture())
-    assertThat(topicProgressResultCaptor.value.isSuccess()).isFalse()
+    val topicProgress = topicProgressResultCaptor.value.getOrThrow()
+    assertThat(topicProgress.storyProgressMap.size).isEqualTo(0)
   }
 
   private fun verifyTopicProgressSucceeded() {
     verify(mockTopicProgressObserver, atLeastOnce()).onChanged(topicProgressResultCaptor.capture())
     assertThat(topicProgressResultCaptor.value.isSuccess()).isTrue()
   }
 
+  private fun verifyStoryProgressFailed(){
+    verify(mockStoryProgressObserver, atLeastOnce()).onChanged(storyProgressResultCaptor.capture())
+    val storyProgress = storyProgressResultCaptor.value.getOrThrow()
+    assertThat(storyProgress.chapterProgressMap.size).isEqualTo(0)
```

Comment:
This doesn't seem like a failure to me--how do we tell the difference between a failure and a default value? Ditto elsewhere.

Reply:
Updated all verifyFailure to actually check failure instead of empty data.